serial.go

```diff
@@ -146,60 +147,118 @@ func (sp *SerialPortList) List() {
 	}
 }
 
-func (sp *SerialPortList) Update() {
-	if !sp.enumerationLock.TryLock() {
-		// already enumerating...
-		return
+// Run is the main loop for port discovery and management
+func (sp *SerialPortList) Run() {
+	for retries := 0; retries < 10; retries++ {
+		sp.runSerialDiscovery()
+
+		logrus.Errorf("Serial discovery stopped working, restarting it in 10 seconds...")
+		time.Sleep(10 * time.Second)
 	}
-	defer sp.enumerationLock.Unlock()
+	logrus.Errorf("Failed restarting serial discovery. Giving up...")
```

Comment:
:thinking: Probably here we should panic/force exit the agent.

Reply:
Forcing the crash/exit of the agent doesn't look like a better alternative...